deepchecks/vision/datasets/detection/coco.py

```diff
@@ -9,28 +9,103 @@
 # ----------------------------------------------------------------------------
 #
 """Module for loading a sample of the COCO dataset and the yolov5s model."""
-import os
+import typing as t
 from pathlib import Path
-from typing import List, Any, Tuple, Optional, Callable
 
 import numpy as np
 import torch
 # import albumentations as A
 from PIL import Image
+from torch import nn
 from torch.utils.data import DataLoader
 from torchvision.datasets import VisionDataset
 from torchvision.datasets.utils import download_and_extract_archive
+from typing_extensions import Literal
 
+from deepchecks import vision
 
-def get_trained_yolov5_object_detection():
+
+__all__ = ['load_dataset']
+
+
+DATA_DIR = Path(__file__).absolute().parent
+
+
+def load_model(pretrained: bool = True) -> nn.Module:
     """Load the yolov5s model and return it."""
-    model = torch.hub.load('ultralytics/yolov5', 'yolov5s', pretrained=True)
+    model = torch.hub.load('ultralytics/yolov5', 'yolov5s', pretrained=pretrained)
     model.eval()
     model.cpu()
-
     return model
 
 
+def load_dataset(
+    train: bool = True,
+    batch_size: int = 32,
+    num_workers: int = 0,
+    shuffle: bool = False,
+    pin_memory: bool = True,
+    object_type: Literal['Dataset', 'DataLoader'] = 'DataLoader'
+) -> t.Union[DataLoader, vision.VisionDataset]:
+    """Get the COCO dataset and return a dataloader.
```

Comment:
I would call it the COCO 128 dataset, as the COCO dataset is much larger than that.

Reply:
Also, did we validate the license of this dataset? and model?

Reply:
changed, 
it is GNU Version 3